pkg/apis/networking/types.go

```diff
@@ -336,19 +393,56 @@ type HTTPIngressRuleValue struct {
 	// options usable by a loadbalancer, like http keep-alive.
 }
 
-// HTTPIngressPath associates a path regex with a backend. Incoming urls matching
-// the path are forwarded to the backend.
+// PathType represents the type of path referred to by a HTTPIngressPath.
+type PathType string
+
+const (
+	// PathTypeExact matches the URL path exactly.
+	PathTypeExact = PathType("Exact")
+
+	// PathTypePrefix matches based on a URL path prefix split by '/'. Matching is
+	// done on a path element by element basis. A path element refers is the
+	// list of labels in the path split by the '/' separator. A request is a
+	// match for path p if every p is an element-wise prefix of p of the
+	// request path. Note that if the last element of the path is a substring
+	// of the last element in request path, it is not a match (e.g. /foo/bar
+	// matches /foo/bar/baz, but does not match /foo/barbaz).
+	PathTypePrefix = PathType("Prefix")
+
+	// PathTypeImplementationSpecific matching is up to the IngressClass.
+	// Interpretation of the Path matching is up to the IngressClass.
+	// Implementations can treat this as a separate PathType or treat it
+	// identically to Prefix or Exact path types.
+	PathTypeImplementationSpecific = PathType("ImplementationSpecific")
+)
+
+// HTTPIngressPath associates a path with a backend. Incoming urls matching the
+// path are forwarded to the backend.
 type HTTPIngressPath struct {
-	// Path is an extended POSIX regex as defined by IEEE Std 1003.1,
-	// (i.e this follows the egrep/unix syntax, not the perl syntax)
-	// matched against the path of an incoming request. Currently it can
-	// contain characters disallowed from the conventional "path"
-	// part of a URL as defined by RFC 3986. Paths must begin with
-	// a '/'. If unspecified, the path defaults to a catch all sending
-	// traffic to the backend.
+	// Path is matched against the path of an incoming request. Currently it can
+	// contain characters disallowed from the conventional "path" part of a URL
+	// as defined by RFC 3986. Paths must begin with a '/'. If unspecified, the
+	// path defaults to a catch all sending traffic to the backend.
```

Comment:
... then all paths are matched

Reply:
Good call, that last line was unclear. I've updated it to read "When unspecified, all paths from incoming requests are matched" and backported those changes back to the older API versions that had the same copy.